lib/ipam/ipam_block.go

```diff
@@ -337,13 +337,19 @@ func (b allocationBlock) attributeRefCounts() map[int]int {
 func (b allocationBlock) attributeIndexesByHandle(handleID string) []int {
 	indexes := []int{}
 	for i, attr := range b.Attributes {
-		if attr.AttrPrimary != nil && *attr.AttrPrimary == handleID {
+		if attr.AttrPrimary != nil && sanitizeHandle(*attr.AttrPrimary) == handleID {
 			indexes = append(indexes, i)
 		}
 	}
 	return indexes
 }
 
+// sanitizeHandle fixes any improperly formatted handles that we might come across (i.e., due to other bugs
+// causing improperly written data.)
+func sanitizeHandle(handleID string) string {
+	return strings.Split(handleID, "\r")[0]
```

Comment:
Since workarounds like this are hard-won, I'd err on the side of a detailed comment:
```
// Work around malformed handles written by host-local to Calico IPAM migration after
// host-local IPAM changed its file format: https://github.com/projectcalico/cni-plugin/issues/821.
```